### ISSUE TITLE
Package and distribute the ASSIST tools as a release artifact

### ISSUE THREAD

**Issue by langston-barrett** [ge-high-assurance/RACK#214]
We currently install the ASSIST tools into the RACK-Box image. This is reasonable, but it would probably be helpful to other TAs to be able to install these tools outside of the Docker image as well, especially `ingest_data`.

**Comment by Ptival** [comment]
Is there more to the assist tools than having the repo checked out and a Prolog interpreter installed?

**Comment by langston-barrett** [comment]
> Is there more to the assist tools than having the repo checked out and a Prolog interpreter installed?

You definitely can use the ASSIST tools by checking out the RACK repo and
installing swi-prolog. This issue is about supporting a more "traditional"
installation for our users that doesn't rely on having access to our source
tree. In particular I'm envisioning this workflow:

Following directions in our documentation,

- User installs `swi-prolog` as a dependency
- User downloads ASSIST tarball from our Github "Releases" page
- User copies files (possibly using a tool like `install`) to appropriate places
  on their system
- User can execute `analyze`, `check`, ... from anywhere on their system and
  expect it to work.

We have another issue open for making sure ASSIST is installed globally inside
the RACK-Box image: https://github.com/ge-high-assurance/RACK/issues/212

**Comment by kquick** [comment]
I think docker provides the ability to execute operations inside the container just like it was a normal application (RUN?), which seems like a convenient way to not require swi-prolog installation or anything like that, but I don't know that using it this way would allow access to the user's local filesystem (it might just be constrained to the in-container filesystem).  If a docker "RUN" is constrained this way, is there anything we could do to facilitate local usage beyond just documenting the process (e.g. yum install, apt package, windows whatever)?

**Comment by langston-barrett** [comment]
> I think docker provides the ability to execute operations inside the container just like it was a normal application (RUN?)

Yes! So this is another point in the design space of "How do users use what we're building?", and it's exactly what issue #212 will enable.

Indeed `docker run` [is super useful](https://docs.docker.com/engine/reference/commandline/run/). For a somewhat technical reason, it's not usable with RACK-box:
```
docker run --rm --entrypoint bash -it interran/rack-box:v3.0 bash 
/usr/bin/bash: /usr/bin/bash: cannot execute binary file
```
An alternative is using `docker exec` to essentially SSH into the container and run it from there, which is viable.

>  I don't know that using it this way would allow access to the user's local filesystem 

`docker run` allows you to mount directories from your host in the container. A downside of `docker run` not working super well (as outlined above) is that you can't specify the mounts for just a single invocation of ASSIST, you have to specify them when you start up RACK (which is a little less flexible).

> If a docker "RUN" is constrained this way, is there anything we could do to facilitate local usage beyond just documenting the process (e.g. yum install, apt package, windows whatever)?

I'm not sure I understand this question, could you rephrase it?
